### PR TITLE
Sending channel and link together generates duplicate text

### PR DESCRIPTION
**Pull request by Bonuspunkt**
#1718

### CODE REVIEW

File: client/js/libs/handlebars/parse.js [hunk 1]
```diff
@@ -66,7 +66,13 @@ module.exports = function parse(text) {
 	const parts = channelParts
 		.concat(linkParts)
 		.concat(emojiParts)
-		.sort((a, b) => a.start - b.start);
+		.sort((a, b) => a.start - b.start || b.end - a.end)
+		.reduce((prev, curr) => {
+			const intersection = prev.some(p => anyIntersection(p, curr));
```
Comment (astorije) [inline client/js/libs/handlebars/parse.js]: `p` ➡️ `(p)`

File: client/js/libs/handlebars/parse.js [hunk 2]
```diff
@@ -66,7 +66,13 @@ module.exports = function parse(text) {
...
+
+			if (intersection) return prev;
```
Comment (astorije) [inline client/js/libs/handlebars/parse.js]: Add `{}` and a new line.

File: client/js/libs/handlebars/parse.js (hunk 1 above)
Comment (astorije) [inline client/js/libs/handlebars/parse.js]: `anyIntersection` is not defined, you need to `require()` it.

**Comment by MaxLeiter** [comment]
Doesn't completely work, `#chan-www.website.com` embeds a website and links as a chan (fine behavior, imo), but the preview arrow doesn't show to collapse. (`/collapse` also doesn't work)
screen shot 2017-11-21 at 8 16 05 pm

**Comment by Bonuspunkt** [comment]
> Also, any chance this could use a new test case? 
no, because it seems you don't really care about it, since previous and upcoming changes doesn't have to include tests

**Comment by xPaw** [comment]
@Bonuspunkt We try to add more tests as we go along, and since you already wrote a test for this it's not a terrible idea to include it.

@MaxLeiter That happens because server uses just `findLinks` and not the full `parse` function, not sure what we should do about that.

**Comment by astorije** [comment]
> no, because it seems you don't really care about it, since previous and upcoming changes doesn't have to include tests

I'm not sure what led you to think that, because we did include tests in changes to the existing code (for example in https://github.com/thelounge/lounge/pull/1446 and https://github.com/thelounge/lounge/pull/1100, the only real 2 changes since your original work). We also try to ask people to add tests when possible, and particularly in this already-well-tested area (I can't find examples, but I know I've asked).

But if you noticed something different, please let us know and we will try to do better!

**Comment by astorije** [comment]
Oh, you do have a test case for it, I didn't know before @xPaw mentioned it: https://github.com/Bonuspunkt/ircmessageparser/pull/12/files#diff-e6c9a6bca996bc454cc244d17bfeda5cR203
@Bonuspunkt, mind bringing it here? :)

**Comment by astorije** [comment]
Hey, closing this PR in favor of https://github.com/thelounge/lounge/pull/1747 to attempt bringing the test. Since you deleted the repo, we can't edit the PR anymore :/
Thanks for your help!!

**Comment by astorije** [comment]
Hey @Bonuspunkt, we have free sticker packs for our contributors now!
If you're interested, go fill the form at https://goo.gl/forms/f5usqAEp5DWoeXC92 🙂

**Pull request closed without merging**